File: 백준/Silver/1541. 잃어버린 괄호/잃어버린 괄호.java
import java.util.*;
import java.lang.*;
import java.io.*;

// The main method must be in a class named "Main".
public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        
        int answer = 0, coefficient = 1, tmp = 0, n = 0;
        String expression = br.readLine();

        expression += "%";
        
        for (int i = 0; i < expression.length(); i++) {
            n = expression.charAt(i) - '0';
            if (n >= 0 && n <= 9) {
                tmp = tmp * 10 + n;
            } else {
                answer += coefficient * tmp;
                tmp = 0;
                if (expression.charAt(i) == '-') coefficient = -1;
            }
        }
        
        System.out.println(answer);
    }
}

File: 백준/Silver/1541. 잃어버린 괄호/잃어버린 괄호.cc
#include <bits/stdc++.h>
#define ll long long

using namespace std;

int main() {
    ios::sync_with_stdio(0);
    cin.tie(0);

    int answer = 0, coefficient = 1, tmp = 0, n = 0;
    string expression;

    cin >> expression;

    expression += '%';

    for (char c : expression) {
        n = c - '0';
        if (n >= 0 && n <= 9) {
            tmp = tmp * 10 + n;
        } else {
            answer += coefficient * tmp;
            tmp = 0;
            if (c == '-') coefficient = -1;
        }
    }

    cout << answer;
    
    return 0;
}